### PR TITLE
Feature/SPUI-111 - Add firebase messaging

### PR DESCRIPTION
**Pull request by VovaMedvediev**

### CODE REVIEW

File: lib/service/services/notification/service/services/register_token_service.dart [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import 'package:dio/dio.dart';
+
+import '../../../../shared/constants.dart';
+import '../../../../shared/helpers/handle_api_responses.dart';
+import '../../model/register_token_request_model.dart';
+
+Future<Map<String, dynamic>> registerTokenService(
+    Dio dio,
+    RegisterTokenRequestModel model,
+    ) async {
```
Comment (eli1stark) [inline lib/service/services/notification/service/services/register_token_service.dart:10]: Formatting

File: lib/shared/services/firebase_messaging_service.dart [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                color: Colors.blue,
+                icon: '@mipmap/ic_launcher',
+              ),
+            ));
+      }
+    });
+  } else {
+    log('User declined or has not accepted permission');
+  }
+}
```
Comment (eli1stark) [inline lib/shared/services/firebase_messaging_service.dart:64]: trailing comas in the file

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/app/screens/navigation/providers/notification_fpod.dart [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import 'notification_spod.dart';
+
+final notificationInitFpod = FutureProvider.autoDispose<void>((ref) async {
+  final notificationService = ref.watch(notificationServicePod);
+  final notification = ref.watch(notificationSpod);
+
+  final token = await FirebaseMessaging.instance.getToken();
+  await registerToken(notificationService, token);
+
+  notification.whenData((token) => registerToken(notificationService, token));
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/screens/navigation/providers/notification_fpod.dart:15]: discussed in dm

File: lib/app/screens/navigation/providers/notification_fpod.dart [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import '../../../../service_providers.dart';
+import '../helpers/register_token.dart';
+import 'notification_spod.dart';
+
+final notificationInitFpod = FutureProvider.autoDispose<void>((ref) async {
+  final notificationService = ref.watch(notificationServicePod);
+  final notification = ref.watch(notificationSpod);
+
+  final token = await FirebaseMessaging.instance.getToken();
+  await registerToken(notificationService, token);
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/screens/navigation/providers/notification_fpod.dart:13]: discussed in dm

File: lib/app/screens/navigation/view/navigation.dart [hunk 5]
```diff
@@ -12,6 +13,9 @@ class Navigation extends HookWidget {
   @override
   Widget build(BuildContext context) {
     final navigation = useProvider(navigationStpod);
+    final notificationInit = useProvider(notificationInitFpod);
+
+    notificationInit.whenData((_) {});
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/screens/navigation/view/navigation.dart:18]: wrap return with notificationInit

File: lib/service/services/notification/service/services/register_token_service.dart (hunk 1 above)
Comment (VovaMedvediev) [reply lib/service/services/notification/service/services/register_token_service.dart:10]: done

File: lib/shared/services/firebase_messaging_service.dart (hunk 2 above)
Comment (VovaMedvediev) [reply lib/shared/services/firebase_messaging_service.dart:69]: done

**Pull request merged**